### PR TITLE
Feature/container store request

### PR DESCRIPTION
**Pull request by bogedal**

### CODE REVIEW

File: src/main/java/dk/bankdata/api/jaxrs/jwt/JwtFilter.java [hunk 1]
```diff
@@ -71,9 +78,8 @@
 @Provider
 @Priority(Priorities.AUTHENTICATION)
 public class JwtFilter implements ContainerRequestFilter {
-
     public static final String JWT_ATTRIBUTE = "JWT";
-    private static final Logger LOG = LoggerFactory.getLogger(JwtFilter.class);
+    private static final Logger LOG = LoggerFactory.getLogger(dk.bankdata.api.jaxrs.jwt.JwtFilter.class);
```
Comment (langecode) [inline src/main/java/dk/bankdata/api/jaxrs/jwt/JwtFilter.java]: Does not need to be fully qualified, does it? :)

File: src/main/java/dk/bankdata/api/jaxrs/jwt/JwtFilter.java [hunk 2]
```diff
@@ -133,8 +139,7 @@ public void filter(ContainerRequestContext requestContext) {
             String jws = authorizationHeader.replace("Bearer ", "");
             JwtClaims jwtClaims = jwtConsumer.processToClaims(jws);
 
-            JwtToken jwtToken = new JwtToken(jwtClaims, jws);
-            requestContext.setProperty(JWT_ATTRIBUTE, jwtToken);
```
Comment (langecode) [inline src/main/java/dk/bankdata/api/jaxrs/jwt/JwtFilter.java]: I think I would go for both. That we have it both on the request context and in the container. I think there are places where we could avoid the container approach, e.g., doing a client filter that passes the token along. So I think I would still place it on the request also...

File: src/main/java/dk/bankdata/api/jaxrs/jwt/JwtFilter.java [hunk 3]
```diff
@@ -169,6 +174,18 @@ public void filter(ContainerRequestContext requestContext) {
         }
     }
 
+    @SuppressWarnings("unchecked")
+    private void storeJwtTokenInContainer(String jws, JwtClaims jwtClaims) {
```
Comment (langecode) [inline src/main/java/dk/bankdata/api/jaxrs/jwt/JwtFilter.java]: :+1:

File: src/main/java/dk/bankdata/api/jaxrs/jwt/JwtTokenContainer.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package dk.bankdata.api.jaxrs.jwt;
+
+import javax.enterprise.context.RequestScoped;
+
+@RequestScoped
+public class JwtTokenContainer {
+    private JwtToken jwtToken;
+
+    public JwtTokenContainer() {}
```
Comment (langecode) [inline src/main/java/dk/bankdata/api/jaxrs/jwt/JwtTokenContainer.java]: Not needed. If you provide no constructor the empty constructor will be implicitly created.

File: src/main/java/dk/bankdata/api/jaxrs/jwt/ValidationException.java [hunk 5]
```diff
@@ -20,7 +20,7 @@ public ProblemDetails getProblemDetails() {
     public boolean equals(Object o) {
```
Comment (langecode) [inline src/main/java/dk/bankdata/api/jaxrs/jwt/ValidationException.java]: Does it make sense to have an equals method on an exception at all?

File: src/main/java/dk/bankdata/api/jaxrs/jwt/JwtFilter.java (hunk 1 above)
Comment (bogedal) [reply src/main/java/dk/bankdata/api/jaxrs/jwt/JwtFilter.java]: Nope - c&p error :)

File: src/main/java/dk/bankdata/api/jaxrs/jwt/JwtTokenContainer.java (hunk 4 above)
Comment (bogedal) [reply src/main/java/dk/bankdata/api/jaxrs/jwt/JwtTokenContainer.java]: :+1:

File: src/main/java/dk/bankdata/api/jaxrs/jwt/ValidationException.java (hunk 5 above)
Comment (bogedal) [reply src/main/java/dk/bankdata/api/jaxrs/jwt/ValidationException.java]: :+1:

File: src/main/java/dk/bankdata/api/jaxrs/jwt/JwtFilter.java (hunk 2 above)
Comment (bogedal) [reply src/main/java/dk/bankdata/api/jaxrs/jwt/JwtFilter.java]: :+1:

**Pull request merged**